src/Sulu/Bundle/PersistenceBundle/DependencyInjection/Compiler/ResolveTargetEntitiesPass.php

```diff
@@ -48,15 +48,20 @@ private function resolve(ContainerBuilder $container)
 
         $resolveTargetEntityListener = $container->findDefinition('doctrine.orm.listeners.resolve_target_entity');
 
+        $interfaceMapping = [];
         foreach ($this->interfaces as $interface => $model) {
+            $interfaceImplementation = $this->getClass($container, $model);
+            $interfaceMapping[$interface] = $interfaceImplementation;
+
             $resolveTargetEntityListener
-                ->addMethodCall('addResolveTargetEntity', [
-                        $interface,
-                        $this->getClass($container, $model),
-                        [],
-                    ]
-                );
+                ->addMethodCall('addResolveTargetEntity', [$interface, $interfaceImplementation, []]);
         }
+
+        // update $targetEntityMapping argument of ReferencesOption service
+        // this is needed to allow for using interfaces when using a "references" option in a doctrine schema
+        $doctrineReference = $container->findDefinition('sulu_core.doctrine.references');
```

Comment:
i guess we should check if `sulu_core.doctrine.references` exists before we access it? 🤔